### ISSUE TITLE
RuboCop CI addition to plots2

### ISSUE THREAD

**Issue by SidharthBansal** [publiclab/plots2#2919]
Let's add RuboCop Continuous Integration to plots2 so that the code is as per the ruby guide.

**Comment by SidharthBansal** [comment]
@siaw23 @jywarren currently we have ``193 rubocop offences``. I think we can solve this in the prs itself after the rubocop ci addition to plots2

**Comment by jywarren** [comment]
I wonder if we could:

1. start with more relaxed rubocop rules so we don't see as many at first (so it's less overwhelming to new people seeing the errors)
2. have it generate `first-timers-only` issues for these, but at some reasonable pace (not all at once as we couldn't be supportive of that many people at once!)

What do you think?

**Comment by jywarren** [comment]
@ryzokuken - you'd been interested in this! I'm also curious. 

@SidharthBansal can you link to the RuboCop CI and also describe how it'd work? 

@siaw23 was mentioning that somehow RuboCop could automatically fix things. I haven't been able to dig out info on that but if there's a link we could look at, that sounds interesting!

**Comment by SidharthBansal** [comment]
Link https://github.com/ad2games/rubocop-ci
Right now I am studying about RuboCop CI. I will be happy to ping you guys once I understand it fully.
Thanks

**Comment by SidharthBansal** [comment]
>You can also use bundle exec rake rubocop:auto_correct to fix most of the issues automatically.

**Comment by siaw23** [comment]
> Link https://github.com/ad2games/rubocop-ci
The person sending a patch needs to do bundle exec rake rubocop:auto_correct to fix most of the issues automatically before committing the changes.
We can add the exceptions in .rubocop_todo.yml file.

my thoughts exactly for a long time now. we should finally get this done. i can single-handedly do this if need be. no automation. Rubocop itself is VERY friendly in pointing out what's wrong, how to fix it.

**Comment by SidharthBansal** [comment]
> i can single-handedly do this if need be.

Ok great.

**Comment by siaw23** [comment]
i'd like to go ahead with this but without the "generating first-timers-only issues" part as i'm not a fan of bots. please let provide thoughts on this. @jywarren

**Comment by jywarren** [comment]
oh cool! single-handedly meaning implementing the cleanup yourself?

I am thinking there are 2 variants we may want to discuss -- A) cleanup of
the current codebase, B) potential to run a code cleanup step on new code
automatically as it's submitted in a PR. Is that right?

Thanks all!!!

**Comment by siaw23** [comment]
yeap. i think creating first-time-issues for this will take forever. so instead i'll do it.

A. yes
B. yes

**Comment by jywarren** [comment]
lol ok!!

**Comment by siaw23** [comment]
by "forever" i meant there are lots of offences/cleanups we have to do, if we had to divide those into small bits it'd take a very long time to finish everything vs a couple of people doing everything in bulk in a short period of time. but i think i can divide them into chunks for first timers, maybe a group of 5 cops at a time and see how it goes... i think it'd be nice to have these easy tasks around to welcome and encourage new contributors.

**Comment by siaw23** [comment]
@SidharthBansal @jywarren 

> Link https://github.com/ad2games/rubocop-ci
The person sending a patch needs to do bundle exec rake rubocop:auto_correct to fix most of the issues automatically before committing the changes.
We can add the exceptions in .rubocop_todo.yml file.

I also think we can encourage contributors to add git pre-commit hooks to "force" users to to run rubocop before they commit

**Comment by SidharthBansal** [comment]
We should encourage them to run rubocop. I agree with this.

As you told that there is a need of system cleanup. So, we should remove the major rubocop errors first. Then the rest errors can be tackled by the new comers. Else it would be annoying for them to deal with the entire repo.

I do agree with your point . We can make multi-step first timers with 5-6 cops.

**Comment by jywarren** [comment]
This sounds great -- once we're ready for FTOs, let's make just one that we can take a look at and see if it's approximately an appropriate amount to work on for a newcomer, before starting to generate a lot. 

Thanks for thinking this through!
